ui/lib/core/addon/components/bar-chart.js

```diff
@@ -0,0 +1,396 @@
+ * @param {function} flattenData - function to flatten object so data isn't nested
+ * @param {array} chartKeys - array of key names associated with the data to display
+ * @param {array} mapLegend - array of objects with key names 'key' and 'label' for the map legend ( i.e. { key: 'non_entity_tokens', label: 'Active direct tokens' })
+ * @param {string} [labelKey=label] - labelKey is the key name in the data that corresponds to the value labeling the y-axis (i.e. "namespace_path" in sampleData)
+ * @param {boolean} [hasExport] - to display export button in top right corner
+ * @param {string} [buttonText=Export data] - text for export button
+ *
+ */
+
+import Component from '@glimmer/component';
+import layout from '../templates/components/bar-chart';
+import { setComponentTemplate } from '@ember/component';
+import { action } from '@ember/object';
+import { guidFor } from '@ember/object/internals';
+import { scaleLinear, scaleBand } from 'd3-scale';
+import { axisLeft } from 'd3-axis';
+import { max } from 'd3-array';
+import { stack } from 'd3-shape';
+// eslint-disable-next-line no-unused-vars
+import { select, event, selectAll } from 'd3-selection';
+// eslint-disable-next-line no-unused-vars
+import { transition } from 'd3-transition';
+
+// SIZING CONSTANTS
+const CHART_MARGIN = { top: 10, left: 137 }; // makes space for y-axis legend
+const CHAR_LIMIT = 18; // character count limit (for label truncating)
+const LINE_HEIGHT = 24; // each bar w/ padding is 24 pixels thick
+
+// COLOR THEME:
+const BAR_COLOR_DEFAULT = ['#BFD4FF', '#8AB1FF'];
+const BAR_COLOR_HOVER = ['#1563FF', '#0F4FD1'];
+const BACKGROUND_BAR_COLOR = '#EBEEF2';
+const TOOLTIP_BACKGROUND = '#525761';
+
+class BarChart extends Component {
+  dataset = [
+    {
+      namespace_id: 'root',
+      namespace_path: 'root',
+      counts: {
+        distinct_entities: 268,
+        non_entity_tokens: 985,
+        clients: 1253,
+      },
+    },
+    {
+      namespace_id: 'O0i4m',
+      namespace_path: 'top-namespace',
+      counts: {
+        distinct_entities: 648,
+        non_entity_tokens: 220,
+        clients: 868,
+      },
+    },
+    {
+      namespace_id: '1oihz',
+      namespace_path: 'anotherNamespace',
+      counts: {
+        distinct_entities: 547,
+        non_entity_tokens: 337,
+        clients: 884,
+      },
+    },
+    {
+      namespace_id: '1oihz',
+      namespace_path: 'someOtherNamespaceawgagawegawgawgawgaweg',
+      counts: {
+        distinct_entities: 807,
+        non_entity_tokens: 234,
+        clients: 1041,
+      },
+    },
+  ];
+
+  get labelKey() {
+    return this.args.labelKey || 'label';
+  }
+
+  get getButtonText() {
+    return this.args.getButtonText || 'Export data';
+  }
+
+  get mapLegend() {
+    return this.args.mapLegend || null;
+  }
+
+  get chartKeys() {
+    return this.args.chartKeys || null;
+  }
+
+  // TODO: take in arguments passed to component
+  flattenedData() {
```

Comment:
Can/should this be a getter?  e.g. `get flattendedData()`?

Reply:
@randallmorey - yes! I agree, my plan was to move this to a parent component that formats the data - which I did this afternoon - so that the `BarChart` just receives an array of objects, along with a legend, to map over and render the data from